ndbench-es-plugins/src/main/java/com/netflix/ndbench/plugin/es/EsWriter.java

```diff
@@ -0,0 +1,191 @@
+
+    private static final ImmutableMap<String, String> EMPTY_PARAMS = ImmutableMap.of();
+    private static final BasicHeader CONTENT_TYPE_HDR_JSON = new BasicHeader("Content-Type", "application/json");
+
+    private final DataGenerator dataGenerator;
+    private final int indexRollsPerDay;
+
+
+    /**
+     * Returns a writer whose {@link EsWriter#writeDocument } method will issue writes to 'esIndexName' and 'esDocType'.
+     *
+     * @param esIndexName        - index name to which writes will be targeted (with possibly appended date pattern,
+     *                           as determined by {@link IEsConfig#getIndexRollsPerDay()}
+     * @param esDocType          - document type (of index named esIndexName) to which writes will be targeted
+     * @param isBulkWrite        - whether to perform 1 write or a batch of writes in the context of a 'writeSingle'
+     *                           call
+     * @param indexRollsPerDay  - a value determined by the configuration setting {@link IEsConfig#getIndexRollsPerDay()}
+     * @param bulkWriteBatchSize - the size the bulk write queue must reach before a bulk write operation is performed
+     *                           and the queue is flushed (ignored if isBulkWrite = false, but nevertheless
+     *                           cannot be less than or equal to zero.)
+     * @param dataGenerator      - data generator used to inject random values into documents written to Elasticsearch.
+     */
+    EsWriter(String esIndexName,
+             String esDocType,
+             boolean isBulkWrite,
+             int indexRollsPerDay,
+             int bulkWriteBatchSize,
+             DataGenerator dataGenerator) {
+        if (bulkWriteBatchSize < 0) {
+            throw new IllegalArgumentException("bulkWriteBatchSize  cannot be less than to zero");
+        }
+        if (!isBulkWrite && indexRollsPerDay > 0) {
+            throw new IllegalArgumentException(
+                    "getIndexRollsPerDay fast property only makes sense to be set when isBulkWrite is set ");
+        }
+
+        this.esIndexUrl = esIndexName + "/" + esDocType;
+        this.esDocType = esDocType;
+        this.indexName = esIndexName;
+        this.bulkWriteBatchSize = bulkWriteBatchSize;
+        this.isBulkWrite = isBulkWrite;
+        this.indexRollsPerDay = indexRollsPerDay;
+        this.dataGenerator = dataGenerator;
+    }
+
+
+    /**
+     * Issues writes to esIndexName' and esDocType given 'restClient' (which determines the host/port of the
+     * Elasticsearch cluster to write to.)
+     */
+    WriteResult writeDocument(RestClient restClient,
+                              String key,
+                              Boolean randomizeKeys) throws Exception {
+        if (isBulkWrite) {
+            writeBatchSizeWorthOfDocs(restClient, key, randomizeKeys);
+        } else {
+            writeSingleDoc(restClient, key, randomizeKeys);
+        }
+
+        return new WriteResult(0);
```

Comment:
Is this supposed to be 0 no matter how many failures there were?

Reply:
Yes. I will try to clarify w/ by returning a constant which has explanatory documentation: 
    /**
     * This constant is provided as a place holder (constant) return value  for 
     * {@link EsWriter#writeDocument(RestClient, String, Boolean)}}, so that for now we effectively ignore 
     * numRejectedExecutionExceptions -- artificially considering them to be zero.  At some future time 
     * these exceptions should affect what write rate we recommend during auto-tuning (i.e., if we see these 
     * exceptions occuring we might want to back off on the rate.)
     */
    public static final WriteResult TMP_PLACEHOLDER_TOTALLY_FINE_RESULT = new WriteResult(0);